### ISSUE TITLE
gradle error about "aaptOptions" when trying to build

### ISSUE THREAD

**Issue by pacoelayudante** [EvilMindDevs/hms-unity-plugin#132]
using 
Unity 2020.3.4
Package 2.0.2
IL2CPP

As soon as I try to create a build, a pop up window shows up, and an error
> mainTemplate.gradle file is using the old aaptOptions noCompress property definition which does not include types defined by unityStreamingAssets constant.

In Unity Preferences I am using all the External Tools that come installed with Unity
I tried removing a segment of the mainGradle as per [this solutions](https://answers.unity.com/questions/1802208/build-error-27.html) but I noticed there is a script in the plugin that overwrites any modification

I have little clue as what to do, I have tried several of the most obvious things

**Comment by sametguzeldev** [comment]
Hello there. When I try to build, I get this pop up and If I click on "Fix and Continue", the build finishes successfully. But nonetheless, we'll be fixing this issue and prevent this pop up as well.
![image](https://user-images.githubusercontent.com/6827857/115062346-ee2af900-9ef2-11eb-8596-1c3c9156217d.png)

**Comment by pacoelayudante** [comment]
Thank you for your quick response, it is very appreciated.

Umm... the popup I see it's different... mostly because it has no "fix" button
![image](https://user-images.githubusercontent.com/9141115/115065935-32070980-9ec5-11eb-9c50-2ddd5006c8f6.png)
there is an interesting difference besides that. The one you show complains about 'launcherTemplate' while the one I see complains about 'mainTemplate'

I am yet to try moving the project to 2020.1 since is the one you are using, I'll see if that fixes the issue for now

as extra data: I don't have any minifier enabled in the project

**Comment by sametguzeldev** [comment]
No worries. We'll be fixing this issue for sure. Probably this weekend :)

**Comment by pacoelayudante** [comment]
In the meantime, if someone is in the same situation
I was able to make it work by commenting the script that overwrites the gradle files, and manually adding the modifications on the gradle files generated by unity
https://github.com/EvilMindDevs/hms-unity-plugin/blob/ed284aa9213a0efe4d91b539213f82f0f1761cdb/Assets/Huawei/Editor/Utils/HMSGradleWorker.cs#L286-L296
I do an early return of this function to avoid the gradle from being overrided

and I add this to the gradle files
> launcherTemplate.gradle
```gradle
...
dependencies {
    implementation project(':unityLibrary')
    implementation 'com.huawei.hms:base:5.2.0.300'
    implementation 'com.android.support:appcompat-v7:28.0.0'
    implementation 'com.huawei.agconnect:agconnect-core:1.4.1.300'
    implementation 'com.huawei.hms:ads-lite:13.4.39.302'
    implementation 'com.huawei.hms:ads-consent:3.4.39.302'
    implementation 'com.huawei.hms:ads-identifier:3.4.39.302'
    }
...
```
> mainTemplate.gradle
```gradle
...
dependencies {
    implementation fileTree(dir: 'libs', include: ['*.jar'])
    implementation 'com.huawei.hms:base:5.2.0.300'
    implementation 'com.android.support:appcompat-v7:28.0.0'
    implementation 'com.huawei.agconnect:agconnect-core:1.4.1.300'
    implementation 'com.huawei.hms:ads-lite:13.4.39.302'
    implementation 'com.huawei.hms:ads-consent:3.4.39.302'
    implementation 'com.huawei.hms:ads-identifier:3.4.39.302'
**DEPS**}
...
```
> baseProjectTemplate.gradle
```gradle
...
repositories {**ARTIFACTORYREPOSITORY**
            google()
            jcenter()
            maven {
                url 'https://developer.huawei.com/repo/'
            }
        }
...
dependencies {
            classpath 'com.android.tools.build:gradle:3.6.0'
            classpath 'com.huawei.agconnect:agcp:1.4.2.300'
            **BUILD_SCRIPT_DEPS**
        }
...
repositories {**ARTIFACTORYREPOSITORY**
        google()
        jcenter()
        flatDir {
            dirs "${project(':unityLibrary').projectDir}/libs"
        }
        maven {
                url 'https://developer.huawei.com/repo/'
        }
    }
...
```

**Comment by sametguzeldev** [comment]
Hello there. Released v2.0.3 version for 2019/2020 unity version and this issue should be resolved in that version :)
